File: _pythonfile/Doit_python/009_[Baekjoon]12891_실버5_DNA 비밀번호.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on('line', function (line) {
  input.push(line);
  if(input.length > 2) rl.close();
}).on('close', function () {
  //console.log(input);
  let [num, DNA, rule] = input;
  //console.log(input);
  let p = Number(num.split(' ')[1]);
  let [a, c, g, t] = rule.split(' ');
  a = Number(a);
  c = Number(c);
  g = Number(g);
  t = Number(t);
  //console.log(a, c, g, t);
  let [start, end] = [0, p];
  let count = 0;

  while (end <= DNA.length) {
    let result = {
      A: 0,
      C: 0,
      G: 0,
      T: 0,
    };
    let temp = DNA.slice(start, end).split('');
    //console.log('temp', temp);
    temp.map((i) => {
      result[i] += 1;
    });
    if (result.A < a || result.C < c || result.G < g || result.T < t) {
      start += 1;
      end += 1;
    } else {
      count += 1;
      start += 1;
      end += 1;
    }
  }
  console.log(count);
  process.exit();
});

File: _pythonfile/Doit_python/009_[Baekjoon]12891_실버5_DNA 비밀번호.py
# 이중 반복문 아니고 => 상태 업데이트에 따라 count 를 올려줘야함
s, p = map(int, input().split())
DNA = input()
a, c, g, t = map(int, input().split())
start, end = [0, p - 1]
result = {"A": 0, "C": 0, "G": 0, "T": 0}
count = 0

# 처음 상태 초기화 a c g t 개수 세기
for i in DNA[start:end + 1]:
  result[i] += 1
# 조건 검사
if result['A'] < a or result['C'] < c or result['G'] < g or result['T'] < t:
  pass
else:
  count += 1
# 상태 업데이트
result[DNA[start]] -= 1
start += 1
end += 1

while end < len(DNA):
  # 상태 업데이트
  result[DNA[end]] += 1
  # 조건 검사
if result['A'] < a or result['C'] < c or result['G'] < g or result['T'] < t:
  pass
else:
  count += 1

result[DNA[start]] -= 1
start += 1
end += 1

print(count)